.github/workflows/reusable-porting-guide.yml

```
@@ -0,0 +1,96 @@
+---
+name: Create porting guide PR
+
+"on":
+  workflow_call:
+    inputs:
+      repository:
+        type: string
+        description: The repository to check out.
+        required: true
+      path:
+        type: string
+        description: The path for the repository.
+        required: true
+      ansible-version:
+        type: string
+        description: Ansible release version.
+        required: true
+      ansible-major-version:
+        type: string
+        description: Ansible major version.
+        required: true
+
+env:
+  git-branch: porting-guide-${{ inputs.ansible-version }}
+  porting-guide-rst: porting_guide_${{ inputs.ansible-major-version }}.rst
+  ci-commit-message: >-
+    Add the Ansible community "${{ inputs.ansible-version }}" porting guide
+  pr-body-message: >-
+    ##### SUMMARY
+
+    Add the Ansible "${{ inputs.ansible-major-version }}" porting guide.
+
+    ##### ISSUE TYPE
+
+    - Docs Pull Request
+
+    ##### COMPONENT NAME
+
+    porting_guide_${{ inputs.ansible-major-version }}.rst
+
+jobs:
+  create-pr:
+    runs-on: ubuntu-latest
+    steps:
```

Since all the steps need the same working dir, you can just change the default.
```suggestion
    defaults:
      working-directory: ${{ inputs.path }}
    steps:
```

---

Added as you suggested following this [Guide.](https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/setting-a-default-shell-and-working-directory)